### DISCUSSION TITLE
How to prepare unit test images for the pathology tumor detection bundle

### DISCUSSION THREAD

**Discussion by yiheng-wang-nv** [Project-MONAI/model-zoo#457] [General]
Preparing images for a bundle's unit test is necessary, currently we produced fake input within the test file like:
https://github.com/Project-MONAI/model-zoo/blob/c2345c18d5e0c78420d8edfca091817107a0d9d6/ci/unit_tests/test_brats_mri_axial_slices_generative_diffusion.py#L60

However, compared with normal images, whole slide images may be more complicated. Hi @drbeh , could you please help to give some suggestions? How to produce fake WSI data for the test? Is there any reference? Thanks in advance!

**Comment by Nic-Ma** [comment]
Hi @yiheng-wang-nv ,

Please refer to how we prepare the test data in MONAI core:
https://github.com/Project-MONAI/MONAI/blob/dev/tests/test_wsireader.py#L39

Thanks.

**Reply by yiheng-wang-nv** [reply]
Thanks @Nic-Ma , I will make use of the same tiff data for test.

**Reply by bhashemian** [reply]
Hi @yiheng-wang-nv, we have multiple tiff files for testing in monai core. So far we are using a generic TIFF file and an Aperio SVS file. Both of them are whole slide images and have pyramidical levels and are actually tiff files but the main difference is metadata. For testing purposes, if you solely need the image data, you can use either of them.

**Comment by yiheng-wang-nv** [comment]
Hi @drbeh , another place to be confirmed is:
https://github.com/Project-MONAI/model-zoo/blob/b4d552e0887fe133adc6a285962217f4bf27d2e2/models/pathology_tumor_detection/configs/metadata.json#L73
"channel_def" should describe what are 3 channels stand for. For example:
https://github.com/Project-MONAI/model-zoo/blob/b4d552e0887fe133adc6a285962217f4bf27d2e2/models/pathology_nuclick_annotation/configs/metadata.json#L64

Could you please help to clarify it? Then I can modify this place in the PR:
https://github.com/Project-MONAI/model-zoo/pull/455

**Reply by bhashemian** [reply]
Hi @yiheng-wang-nv, for tumor detection bundles channels, I confirm that the channel_def is as follow:
```
"0": "R",
"1": "G",
"2": "B"
```
